pyleecan/Methods/Slot/Hole/comp_magnetization_dict.py

```diff
@@ -0,0 +1,36 @@
+from numpy import pi
+
+
+def comp_magnetization_dict(self, is_north=True, return_type=2):
+    """Compute the dictionary of the magnetization direction of the magnets (key=magnet_X, value=angle[rad])
+    Mangetization angle with Hole centered on Ox axis
+
+    Parameters
+    ----------
+    self : Hole
+        a Hole object
+    is_north: True
+        True: comp north magnetization, else add pi [rad]
+    return_type : int
+        0: Normal as tuple of complex
+        1: Z2 - Z1 (with Z1 and Z2 the points from return_type 0)
+        2: the angle of the vector of return_type 1 according to Ox [rad]
+
+    Returns
+    -------
+    mag_dict: dict
+        magnetization dictionary (key=magnet_X, value=angle[rad])
+    """
+
+    if self.magnetization_dict_enforced is not None:
+        if return_type != 2:
+            raise Exception(
+                "For Hole, only return_type=2 (float) is available for comp_magnetization_dict"
+            )
+        mag_dict = self.magnetization_dict_enforced.copy()
+        if not is_north:
+            for key in mag_dict.keys():
+                mag_dict[key] += pi
+        return mag_dict
```

Comment:
This seems inconsistent with HoleMxx, since there is modification of the enforced values, although this is the abstracts class method only.

Reply:
Indeed. Do you think that we should say that magnetization_dict_enforced enforce the "north direction" and apply this modification in all HoleMx ? The other solution is maybe to remove this option (I think it is unused anyway). What do you prefer ?

Reply:
I think this shouldn't be removed but disabled for now. 
We should discuss in a dedicated issue, what this feature can do, e.g. also utilize the FEMM magnetization by string feature (see Hallbach). Further it should be avialable for surface magnets also.

Reply:
I was talking about the "is_north" option ;) Indeed the magnetization_dict_enforced  will be quite helpful in the long run. So I will comment the is_north part in all holes for now.

Reply:
So regarding enforced values, the question is, will it be absolute values or offset to the computed. I think offset would be more accessible, since the user dont have to care about the actual magnets position.

Reply:
I think that we should either use "absolute definition " either change the property name to "magnetization_dict_offset". 
With "absolute definition", we can create the offset by calling the comp_magnetization_dict and updating this dictionary. Then for HoleUD and all magnets we take directly the value within the dict. 
With the "offset definition", we can say that the "computed part" of HoleUD is 0 and enforce the offset. The offset definition is interesting when considering sensitivity/fault study since "0" would be the reference direction. I agree that we will need to extend this definition to surface magnet and one day we may also want to be able to change the magnetization any of the "Zs" magnets (by using Magnet_R0_T0_S0 notation maybe). For this case again, "offset definition" would be more convenient. So I will update the code to the "offset definition".

Reply:
Then with the "offset definition", we can use "is_north" to add a "pi offset" ;)